### DISCUSSION TITLE
calculator not accurate : 71% instead of 71.04% value

### DISCUSSION THREAD

**Discussion by enrack** [raylu/poecalc#18] [General]
Hi, it seems that the project only use int values instead of double.
I have a 0.04% difference on every auras i use :/ It's huge when it comes to calculate 300M+ dps.
Can you fix it ? it's easy i think.
Thanks

**Comment by raylu** [comment]
can you give a more specific example of where the calcs are wrong? an account/character name would be best

**Reply by enrack** [reply]
Hi,
Account : moineshoo
Name : focusmpfive
Aura-effect : 176 (when i load without puting any value there, it says 142%, while pob says i got 176% on my weakest auras, so i put it manualy).

So, if we take the defiance banner for example (the easiest aura to calculate) : 
The calculator says I got 196% increase aura-effect on it (which is exactly what pob says, no problem).
Defiance banner has a base value of : "24% increased armour and evasion rating" at lvl 20.
So if we calculate mannualy the true defiance banner value after aura-effect, it should be : 24 x 2.96 = 71.04 .
The calculator says that my defiance banner gives 71% increase armour, which is false, it gives 71.04% armour. And that's why my spark carry mate has a bigger armour value in-game than on the pob (after pasting the values in config tab).
And that difference is pretty much everywhere. So when it comes to estimate big armour, or big crit multiplier, or big dps, the difference can be pretty huge.
For example the pob says 85% phys. reduction instead of 90% (because the same difference occurs on the divergent determination for example, and maybe else-where).

So in my opinion, the problem is when doing the calculation : 
base_skill_value x (1 + related_increased_aura_effect_value / 100)
example with defiance banner :
24 x (1 + 196 / 100)
The result of that last formula should be 71.04%, and not 71% like the calculator says. So I should put 71.4% increased armour and evasion in config tab on pob ; Not 71%

I think it's just a variable type matter, easily fixable, right ?
The real challenge is : how to put 71.04% on config tab in pob, if (and tell me if i'm mistaken) pob doesn't accept the .04
Maybe there's a way of bypassing this ? I'm not a pob expert.

Thanks a lot, your calculator is still amazing, way more accurate, simple, and portable than the other ones I came across.
Very good job !

Thanks for the reply :)

**Comment by raylu** [comment]
>  The real challenge is : how to put 71.04% on config tab in pob, if (and tell me if i'm mistaken) pob doesn't accept the .04
> Maybe there's a way of bypassing this ? I'm not a pob expert.

right. when I tested this, PoB also didn't like any non-whole-number increases. I believe the relevant code is https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/31798a35f7f0deb87e5a9be3939737d6c92d1060/src/Modules/ModParser.lua#L41 which only handles whole numbers. you could argue this is a bug in PoB (though the mod parser is really just to parse mods that could appear on gems, the skill tree, and items, and I don't think there are any increases more granular than 1% there)

that's why we intentionally floor the values here https://github.com/raylu/poecalc/blob/0337fac1cefdbf0d76c0be147f4fff642367fbb5/gems.py#L419 to get PoB to parse them
